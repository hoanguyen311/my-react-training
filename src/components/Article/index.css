.article {
    display: flex;
    text-decoration: none;
    background: #FFF;
}

.article__image-wrap {
    display: flex;
    align-items: center;
}

.article__image {
    width: 100%;
    height: auto
}

.article__info {

}

.article__title {
    font-size: 15px;
    color: #ccc;
}

.article__author {}
.article__description {}
