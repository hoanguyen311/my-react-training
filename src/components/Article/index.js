import React from 'react';
import PropsType from 'prop-types';
import './index.css';

function Article(props) {
    const { author, title, description, url, urlToImage, publishedAt } = props;

    const info = (
        <div className="article__info">
            <h2 className="article__title">{title}</h2>
            <h2 className="article__title">{title}</h2>
            <b className="article__author">{author}</b>
            <p className="article__description">{description}</p>
            <p className="article___time">{publishedAt}</p>
        </div>
    );

    const image = (
        <span className="article__image-wrap">
            <img src={urlToImage} alt={title} className="article__image"/>
        </span>
    );

    if (url) {
        return (
            <a href={url} className="article article_type_link" target="_blank">
                {image}
                {info}
            </a>
        )
    }

    return (
        <div className="article">
            {image}
            {info}
        </div>
    );
}

Article.propTypes = {
    title: PropsType.string.isRequired,
    author: PropsType.string,
    description: PropsType.string,
    url: PropsType.string,
    urlToImage: PropsType.string,
    publishedAt: PropsType.string
};

export default Article;
