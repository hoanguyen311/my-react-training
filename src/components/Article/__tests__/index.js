import React from 'react';
import Article from '../.';
import renderer from 'react-test-renderer';

const mock = {
    "author": "Jon Russell",
    "title": "In India, an Uber for farm machinery aims to make a difference in rural areas",
    "description": "Uber has inspired countless businesses to adopt its asset-light and on-demand approach to their industries. The examples are countless. Food delivery, dry..",
    "url": "https://techcrunch.com/2017/08/15/em3/",
    "urlToImage": "https://tctechcrunch2011.files.wordpress.com/2017/08/em3.jpg?w=764&h=400&crop=1",
    "publishedAt": "2017-08-15T09:58:59Z"
};

const mockWithoutUrl = {
    "author": "Jon Russell",
    "title": "In India, an Uber for farm machinery aims to make a difference in rural areas",
    "description": "Uber has inspired countless businesses to adopt its asset-light and on-demand approach to their industries. The examples are countless. Food delivery, dry..",
    "urlToImage": "https://tctechcrunch2011.files.wordpress.com/2017/08/em3.jpg?w=764&h=400&crop=1",
    "publishedAt": "2017-08-15T09:58:59Z"
};

describe('test aritcle by snapshot', () => {
    it('render correctly with A tag', () => {
        const article = renderer.create(
            <Article {...mock} />
        ).toJSON();

        expect(article).toMatchSnapshot();
    });

    it('render correctly without A tag', () => {
        const article = renderer.create(
            <Article {...mockWithoutUrl} />
        ).toJSON();

        expect(article).toMatchSnapshot();
    });
});
