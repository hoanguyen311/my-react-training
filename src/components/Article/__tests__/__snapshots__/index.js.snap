// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test aritcle by snapshot render correctly with A tag 1`] = `
<a
  className="article article_type_link"
  href="https://techcrunch.com/2017/08/15/em3/"
  target="_blank"
>
  <span
    className="article__image-wrap"
  >
    <img
      alt="In India, an Uber for farm machinery aims to make a difference in rural areas"
      className="article__image"
      src="https://tctechcrunch2011.files.wordpress.com/2017/08/em3.jpg?w=764&h=400&crop=1"
    />
  </span>
  <div
    className="article__info"
  >
    <h2
      className="article__title"
    >
      In India, an Uber for farm machinery aims to make a difference in rural areas
    </h2>
    <h2
      className="article__title"
    >
      In India, an Uber for farm machinery aims to make a difference in rural areas
    </h2>
    <b
      className="article__author"
    >
      Jon Russell
    </b>
    <p
      className="article__description"
    >
      Uber has inspired countless businesses to adopt its asset-light and on-demand approach to their industries. The examples are countless. Food delivery, dry..
    </p>
    <p
      className="article___time"
    >
      2017-08-15T09:58:59Z
    </p>
  </div>
</a>
`;

exports[`test aritcle by snapshot render correctly without A tag 1`] = `
<div
  className="article"
>
  <span
    className="article__image-wrap"
  >
    <img
      alt="In India, an Uber for farm machinery aims to make a difference in rural areas"
      className="article__image"
      src="https://tctechcrunch2011.files.wordpress.com/2017/08/em3.jpg?w=764&h=400&crop=1"
    />
  </span>
  <div
    className="article__info"
  >
    <h2
      className="article__title"
    >
      In India, an Uber for farm machinery aims to make a difference in rural areas
    </h2>
    <h2
      className="article__title"
    >
      In India, an Uber for farm machinery aims to make a difference in rural areas
    </h2>
    <b
      className="article__author"
    >
      Jon Russell
    </b>
    <p
      className="article__description"
    >
      Uber has inspired countless businesses to adopt its asset-light and on-demand approach to their industries. The examples are countless. Food delivery, dry..
    </p>
    <p
      className="article___time"
    >
      2017-08-15T09:58:59Z
    </p>
  </div>
</div>
`;
